sdk/spring/azure-spring-boot-samples/azure-spring-boot-sample-active-directory-b2c-resource-server/README.md

```diff
@@ -0,0 +1,117 @@
+# Sample for Azure AD B2C Resource server Spring Boot client library for Java
+
+## Key concepts
+This sample illustrates how to use `azure-spring-boot-starter-active-directory-b2c` package to work with OAuth 2.0 and OpenID Connect protocols with Azure AD B2C to secure web services.
+
+## Getting started
+
+### Environment checklist
+We need to ensure that this [environment checklist][ready-to-run-checklist] is completed before the run.
+
+### Create and consent Application permissions 
+1. On the **Azure AD B2C** Portal, select the application that requires roles to be added, select **Manifest**.
+2. Find the `appRoles` configuration item, and add the following configuration, then click the **Save** button.
+```json
+  {
+    "allowedMemberTypes": [
+      "Application"
+    ],
+    "description": "Task.read",
+    "displayName": "Task.read",
+    "id": "d2bec026-b75f-418d-9493-8462f54f25d9",
+    "isEnabled": true, 
+    "value": "Test.read"
+  },
+  {
+    "allowedMemberTypes": [
+      "Application"
+    ],
+    "description": "Task.wirte",
+    "displayName": "Task.wirte",
+    "id": "1ab4eeda-d07e-4bce-8f77-b0a84c97c34f",
+    "isEnabled": true,
+    "value": "Test.wirte"
+  }
+```
+
+![Configuration Application Roles](docs/image-configuration-application-roles.png "Configuration Application Roles")
+
+3. Find the application permissions need to use.
+
+![Selected Application](docs/image-selected-application.png "Selected Application")
+![Add Application Roles](docs/image-add-application-roles.png "Add Application Roles")
+
+4. Consent Application permissions.
+
+![Consent Application permissions](docs/image-consent-application-permissions.png "Consent Application permissions")
+
+5. In the end, configuration is as follows.
+   
+![Final Configuration](docs/image-final-configuration.png "Final Configuration")
+
+## Examples
+### Configure the sample
+#### application.yml
+
+```yaml
+# In v2.0 tokens, this is always the client ID of the API, while in v1.0 tokens it can be the resource URI used in the request.
```

Comment:
We should be more accurate about this, 

> In v2.0 tokens, this is always the client ID of the API, while in v1.0 tokens it can be the resource URI used in the request.

`this` stands for what?